app/(gcforms)/[locale]/(form filler)/id/[...props]/lib/notifications.ts

```diff
@@ -0,0 +1,246 @@
+import { sendEmail } from "@lib/integration/notifyConnector";
+import { logMessage } from "@lib/logger";
+import { getRedisInstance } from "@lib/integration/redisConnector";
+import { getOrigin } from "@lib/origin";
+import { NotificationsInterval } from "@gcforms/types";
+import { serverTranslation } from "@i18n";
+import { prisma, prismaErrors } from "@lib/integration/prismaConnector";
+import { logEvent } from "@lib/auditLogs";
+
+export const Status = {
+  SINGLE_EMAIL_SENT: "SINGLE_EMAIL_SENT",
+  MULTIPLE_EMAIL_SENT: "MULTIPLE_EMAIL_SENT",
+} as const;
+export type Status = (typeof Status)[keyof typeof Status];
+
+// Sends an email notification when a user has new form submissions
+export const sendNotification = async (formId: string, titleEn: string, titleFr: string) => {
+  const usersAndNotifications = await getUsersAndNotificationsInterval(formId);
+  if (!usersAndNotifications) {
+    logMessage.error(`sendNotification template not found with id ${formId}`);
+    return;
+  }
+
+  const { users } = usersAndNotifications;
+
+  const notificationsInterval =
+    usersAndNotifications.notificationsInterval as NotificationsInterval;
+  if (!validateNotificationsInterval(notificationsInterval)) {
+    logMessage.error(
+      `sendNotification template ${formId} has an invalid notificationsInterval ${notificationsInterval}`
+    );
+    return;
+  }
+
+  // Notifcations are turned off, do nothing
+  if (!notificationsInterval) {
+    return;
+  }
+
+  const marker = await getMarker(formId);
+  switch (marker) {
+    case Status.SINGLE_EMAIL_SENT:
+      // Single submissions email sent but not multiple submissions email, send multiple email
+      Promise.all([
+        sendEmailNotificationsToAllUsers(users, formId, titleEn, titleFr, true),
+        setMarker(formId, notificationsInterval, Status.MULTIPLE_EMAIL_SENT),
+      ]);
+      break;
+    case Status.MULTIPLE_EMAIL_SENT:
+      // Multiple submissions email has been sent, do nothing
+      break;
+    default:
+      // No email has been sent, send single submission email
+      Promise.all([
+        sendEmailNotificationsToAllUsers(users, formId, titleEn, titleFr, false),
+        setMarker(formId, notificationsInterval),
+      ]);
+  }
+};
+
+export const validateNotificationsInterval = (notificationsInterval: NotificationsInterval) => {
+  let isValid = false;
+  Object.values(NotificationsInterval).forEach((interval) => {
+    if (notificationsInterval !== interval) {
+      isValid = true;
+    }
+  });
+  return isValid;
+};
+
+async function getUsersAndNotificationsInterval(formID: string): Promise<{
```

Comment:
perhaps prefix with an underscore to make it quickly known that this is an internal function that should not be exported.